### PR TITLE
Split PortableThreadPool.WorkerThread start and loop body

### PR DESCRIPTION
**Pull request by lambdageek**
This is Part 1 of https://github.com/dotnet/runtime/issues/84489 - landing support for async JS interop on threadpool threads in multi-threaded WebAssembly.

We will need to start the threadpool worker threads on the browser in a special way, such that they can exit back to the JS event loop, and use callbacks to run the worker loop body.

The current PR splits into a separate file the logic for starting threadpool worker threads, and the outer loop that waits for the semaphore that signals that work is available for the worker.  The loop body (to be shared with the callback-based approach in a future PR) remains in `PortableThreadPool.WorkerThread.cs` as several new toplevel functions.

Current PR is just refactoring existing code. No functional change.

### CODE REVIEW

**Comment by lambdageek** [comment]
For reviewers:  I recommend turning on "hide whitespace changes" for this PR

Screenshot 2023-04-07 at 16 06 01

**Comment by lambdageek** [comment]
@kouvel could you take a look

File: src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.NonBrowser.cs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+                LowLevelLock threadAdjustmentLock = threadPoolInstance._threadAdjustmentLock;
+                LowLevelLifoSemaphore semaphore = s_semaphore;
+
+                while (true)
+                {
+                    bool spinWait = true;
+                    while (semaphore.Wait(ThreadPoolThreadTimeoutMs, spinWait))
+                    {
+                            WorkerDoWork(threadPoolInstance, ref spinWait);
```
Comment (kouvel) [inline src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.NonBrowser.cs:58]: Indentation here and below at the break is a bit off

**Review by kouvel — APPROVED** [review]
Just a couple of minor suggestions, otherwise LGTM

File: src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs [hunk 2]
```diff
@@ -28,148 +29,115 @@ private static partial class WorkerThread
...
+                    RemoveWorkingWorker(threadPoolInstance);
                 }
+            }
 
-                LowLevelLock threadAdjustmentLock = threadPoolInstance._threadAdjustmentLock;
-                LowLevelLifoSemaphore semaphore = s_semaphore;
+            // returns true if the worker is shutting down
+            // returns false if we should do another iteration
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            private static bool WorkerTimedOutMaybeStop (PortableThreadPool threadPoolInstance, LowLevelLock threadAdjustmentLock)
```
Comment (kouvel) [inline src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WorkerThread.cs:83]: Not sure about the name, how about `TryExitWorker` instead?

**Comment by lambdageek** [comment]
Failure is https://github.com/dotnet/runtime/issues/84979

**Pull request merged**